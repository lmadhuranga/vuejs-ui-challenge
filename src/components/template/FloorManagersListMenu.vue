<template id="floorManagersListMenu">
  <ul>
    <li>
      <p><router-link :to="{ path: `/setup/userAdd`}">Add Project Manager</router-link></p>
      <small>Loet 6, Schagen</small>
    </li>
    <li v-for="floorManager in filteredFloorManagers"  >
      <p><router-link v-bind:key="floorManager.id" :to="{ path: `/setup/shiftAdd/${floorManager.id}`}">{{floorManager.name}}</router-link></p>
      <small>Stationsweg 1, Breukelen</small>
    </li>
  </ul>
</template>

<script>
  export default {
    components: {},
    data(){
      return {
        search: ''
      }
    },
    name: 'FloorManagersListMenu',
    props: ['floorManagers'],
    computed: {
      filteredFloorManagers() {
        return this.floorManagers.filter(floorManager => {
          return floorManager.name.indexOf(this.search.toLowerCase()) > -1
        })
      }
    }
  }
</script>
