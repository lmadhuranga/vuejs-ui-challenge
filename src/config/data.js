export default {
  users: [
    {
      id: 1,
      name: 'Madhuranga Sendheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
    {
      id: 2,
      name: 'Randika Senadheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
  ],
  projects: {
    1: {
      id: 1,
      name: 'Project 1',
      shifts: [
        {
          id: 1,
          name: 'shift1',
          iban: 'IbanDetails',
          description: 'description',
          userId: 1
        },
        {
          id: 2,
          name: 'shift2',
          iban: 'IbanDetails',
          description: 'description',
          userId: 2
        }
      ]
    },
    2: {
      id: 1,
      name: 'Project 2'
    },
    3: {
      id: 1,
      name: 'Project 3'
    }
  }
}
