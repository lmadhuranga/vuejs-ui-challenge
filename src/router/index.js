import Vue from 'vue'
import Router from 'vue-router'
import Home from '@/components/pages/HomePage'
import UserAdd from '@/components/pages/UserAdd'
import UserView from '@/components/pages/UserView'
import UsersList from '@/components/pages/UsersList'
import SetupContainer from '@/components/pages/SetupContainer'
import Vuetify from 'vuetify'

Vue.use(Vuetify);
Vue.use(Router);

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(plugin)
}

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home
    },
    {
      path: '/userslist',
      name: 'usersList',
      component: UsersList
    },
    {
      path: '/setup',
      name: 'setup',
      component: SetupContainer,
      children: [
        {
          // UserProfile will be rendered inside User's <router-view>
          // when /user/:id/profile is matched
          path: 'add',
          component: UserAdd
        }]
    },
    {
      path: '/add',
      name: 'add',
      component: UserAdd
    },
    {
      path: '/:id',
      name: 'userView',
      component: UserView
    }
  ]
})

