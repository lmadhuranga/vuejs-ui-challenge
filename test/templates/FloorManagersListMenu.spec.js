import Vue from 'Vue';
import FloorManagersListMenu from '../../src/components/template/FloorManagersListMenu.vue'
import router from '../../src/router'
import data from '../../src/config/data'
import {shallow, mount} from "vue-test-utils"
import VueRouter from 'vue-router';

describe('FloorMangersListMenu.vue', () => {
  Vue.use(VueRouter);
  let cmp, floorManager;

  const createCmp = propsData => shallow(FloorManagersListMenu, {
    propsData,
    el: document.createElement('div'),
    router: router,
    render: h => h('router-view')
  });

  beforeEach(() => {
    floorManager = data.users[0];

    cmp = createCmp({
      floorManagers: [floorManager]
    });
  });

  it('should be able load floorManagers', () => {
    expect(cmp.vm.floorManagers[0]).toEqual(floorManager);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  });
});
