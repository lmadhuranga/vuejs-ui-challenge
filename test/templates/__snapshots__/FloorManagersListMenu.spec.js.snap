// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorManagersListMenu.vue has the expected html structure 1`] = `<!--  -->`;
